docs (auth/ldap): add new fields maximum_page_size, dereference_aliases (#3700)
hashicorp/boundary: website/content/docs/concepts/domain-model/auth-methods.mdx

  the value equals the `to_attribute`.  For example, `preferredName=fullName`.  All
  attribute names are case insensitive.

  - `maximum_page_size` - (optional) If set, it specifies a maximum ldap search
    result size to use when retrieving the authenticated user's group

>     memberships. This can be used to avoid reaching the LDAP server's max result

Reviewer: ```suggestion
    memberships. You can use this setting to avoid reaching the LDAP server's max result
```

    size. 

  - `dereference_aliases` - (optional) If set, it will control how aliases are

>     dereferenced when performing the search.

Reviewer: ```suggestion
    dereferenced when you search.
```
